ecdsa/src/ecdsa.rs

```diff
@@ -131,9 +133,18 @@ impl<E: CurveAffine, N: PrimeField, const NUMBER_OF_LIMBS: usize, const BIT_LEN_
         let q_x_reduced_in_r = scalar_chip.reduce_external(ctx, &q_x_reduced_in_q)?;
 
         // 7. check if Q.x == r (mod n)
-        scalar_chip.assert_strict_equal(ctx, &q_x_reduced_in_r, &sig.r)?;
-
-        Ok(())
+        let is_q_x_reduced_in_r_equal_to_r = scalar_chip.is_strict_equal(ctx, &q_x_reduced_in_r, &sig.r)?;
+        
+        let is_valid = scalar_chip.and(ctx, &is_r_s_valid, &is_q_x_reduced_in_r_equal_to_r)?;
+        let value_1 = enable_skipping_invalid_signature as u64;
+        let enable_skipping_invalid_signature_value = scalar_chip.assign_constant(ctx, value_1.into())?;
```

Comment:
`value_1` a bit of confusing name. Would either just do the conversion in the `assign_constant` or just overwrite `enable_skipping_invalid_signature`.

Reply:
Updated. Overwrite ```enable_skipping_invalid_signature```